### PR TITLE
[WIP] Fixed comments, user GET, refactoring, pylint

### PR DESCRIPTION
**Pull request by Krisselchen**
fixed message field in manifest.
Furthermore fixed some creepy bugs in comments.
I am not happy with line 268 to 270: 
The only solution I had, would be, to initialize comments in projects POST for every manifest with an empty array in comments field to distinguish between no projects found an no comments found. Of course comments have to be required in scheme in this case. Any different ideas?

### CODE REVIEW

File: backend/api/projects.py
Comment (henderkes) [inline backend/api/projects.py]: shouldn't a user be able to delete or edit any comments under his projects, even if they aren't his own?

Comment (henderkes) [reply backend/api/projects.py]: @Krisselchen answered this, it's working as intended here

**Review by henderkes — DISMISSED** [review]
approved from me, please wait for a second approval because many changes were from me too

**Comment by henderkes** [comment]
Mobile github site is the best...

**Comment by henderkes** [comment]
this fixes #250, #251, #265, #270, #279, #283, #290

yeah.. a bit much for one pr, I'm sorry

**Comment by gitmirgut** [comment]
`POST` '/api/projects/<uuid:project_id>/comment' not working.
message:
```
{
    "message": "name 'user' is not defined",
    "validation_error": null
}
```

**Comment by henderkes** [comment]
hotfixed it, find&replace for user->current_user didn't work because there was no space before it after a + operator

**Comment by gitmirgut** [comment]
POST '/api/projects/uuid:project_id/comment' not still working.
message:
```
{
    "message": "maximum recursion depth exceeded in comparison",
    "validation_error": null
}
```

File: backend/api/projects.py
Comment (henderkes) [inline backend/api/projects.py:321]: we should adjust the validator long term, this is a poor solution but works for now

File: backend/api/projects.py
Comment (henderkes) [inline backend/api/projects.py:404]: I think a validation error shouldn't return 500

File: backend/api/search.py
Comment (henderkes) [inline backend/api/search.py]: should probably add exception handling

File: backend/api/users.py
Comment (henderkes) [inline backend/api/users.py:47]: exception handling too maybe

**Review by henderkes — APPROVED** [review]
A couple TODOs, approved otherwise

File: backend/api/projects.py
Comment (Krisselchen) [reply backend/api/projects.py:404]: But the validation Error occurs by the backend. The projectmanifest is only used internally here. I am not really sure about it.

File: backend/api/search.py
Comment (Krisselchen) [reply backend/api/search.py]: Shouldn't it be in projects.py?

Comment (henderkes) [reply backend/api/search.py]: I'm not sure, felt like it's more of a search thing but we can move it to projects

File: backend/api/users.py
Comment (henderkes) [inline backend/api/users.py]: needs an explicit str(id) cast

**Review by henderkes — CHANGES_REQUESTED** [review]

File: backend/api/projects.py
Comment (henderkes) [inline backend/api/projects.py]: shouldn't we return a list of author dicts instead? I mean we can sort them this way, but returning names first and emails after with no real distinction doesn't seem a good idea

**Review by henderkes — CHANGES_REQUESTED** [review]

**Review by henderkes — DISMISSED** [review]

File: backend/api/users.py
Comment (mharasic) [reply backend/api/users.py]: with cast its not working unfortunately

Comment (henderkes) [reply backend/api/users.py]: are you sure? it's not working without a cast because it can't implicitly convert uuid to str, but explicit conversion from uuid to str works just fine

Comment (Drakulix) [reply backend/api/users.py]: What means not working?

File: backend/api/projects.py
Comment (henderkes) [reply backend/api/projects.py:404]: this is the DELETE call so I was mistaken in the first place, 500 is correct and PUT and POST were already returning correct codes

File: backend/api/users.py
Comment (henderkes) [reply backend/api/users.py]: it raised a 500 error because flask didn't handle the exception (because it can't implitly cast). 

changed it to str(id) now and I don't see a reason why it wouldn't work now, it's only the response text anyways.

**Review by henderkes — DISMISSED** [review]
More reviews please.

**Review by Drakulix — APPROVED** [review]

**Review by henderkes — APPROVED** [review]

**Review by Leats — APPROVED** [review]

**Pull request merged**
